### ISSUE TITLE
MKL library on Windows

### ISSUE THREAD

**Issue by 0936487** [evalf/nutils#718]
We experienced issues with importing the MKL library on Windows for Python version >= 3.8. The problem is two-fold:
1. The library was renamed, now called mkl_rt.2.dll (previously mkl_rt.dll) 
2. The directory of the library needs to be imported explicitly using os.add_dll_directory("Path to dll-dependencies for your dll library")

The following (pseudo) code would fix this problem in _mkl.py:

```python
if python_version >= 3.8:
  os.add_dll_directory(MKL_DIRECTORY)
  libname='mkl_rt.2.dll'
else:
  libname='mkl_rt.dll'
```

followed by the import line which is already there with the filename replaced by libname for the win32 case.

**Comment by joostvanzwieten** [comment]
In your pseudo code what is `MKL_DIRECTORY`?

**Comment by 0936487** [comment]
The directory where the file 'mkl_rt.2.dll' is located. I am not sure whether this is the same directory on all machines.

**Comment by joostvanzwieten** [comment]
If the location of the MKL libraries is not fixed, I'm affraid we can't provide a solution to the second problem within Nutils. On Linux machines you can add library search directories to environment variable `LD_LIBRARY_PATH`. In Windows you can do the same using [environment variable `PATH`](https://learn.microsoft.com/nl-nl/windows/win32/dlls/dynamic-link-library-search-order#standard-search-order-for-desktop-applications).

Can you try the following? Revert the fix for the second problem, but keep `libname='mkl_rt.2.dll'`. Open a terminal and run

```sh
set PATH=%PATH%;MKL_DIRECTORY
```

where `MKL_DIRECTORY` is the directory where the MKL libraries are installed. Then run you script *in the same terminal*. The change to `PATH` applies to this terminal only. You can make the change permanent, but I'm not sure how.

**Comment by joostvanzwieten** [comment]
@0936487 Does it work with the environment variable?

PR #723 fixes the first problem. Would you be able to test this?

**Comment by 0936487** [comment]
Update from my side: 
- The branch `support-multiple-mkl-versions` does work as it is now able to find a different version of the `.mkl `file
- The `%PATH%` variable does also not work on a non-virtual Windows installation. For now, we still have to specify the directory manually (`MKL `was installed using pip) using the `os.add_dll_directory()`

**Comment by joostvanzwieten** [comment]
Thanks for the update. I've merged #723 (backport: #729) and closed this issue because I think that the PR provides a sufficient solution to the issue. If you disagree, please reopen the issue.

**Comment by joostvanzwieten** [comment]
The backport is included in Nutils version 7.2, which has just been released.
